src/util.rs

```diff
@@ -79,3 +79,10 @@ pub fn duration_from_timeval(ts: timespec) -> Duration {
 pub fn system_time_from_timespec(ts: timespec) -> SystemTime {
     UNIX_EPOCH + duration_from_timeval(ts)
 }
+
+pub fn hal_id_to_raw(id: Id) -> u32 {
```

Comment:
I wonder if this isn't something we could push upstream to the embedded-hal/can crate?
```
impl Id {
    pub fn as_raw(&self) -> u32 {
        match self {
            Id::Standard(id) => id.as_raw() as u32,
            Id::Extended(id) => id.as_raw(),
        }
    }
}
```

Reply:
I threw a PR at them. Let's see how it goes.
https://github.com/rust-embedded/embedded-hal/pull/428